.dotnet/src/Custom/Chat/ChatResponseFormat.cs

```diff
@@ -1,45 +1,71 @@
-using OpenAI.Models;
+using OpenAI.Internal;
+using System;
+using System.ClientModel.Primitives;
+using System.ComponentModel;
 
 namespace OpenAI.Chat;
 
-/// <summary>
-/// Represents a requested <c>response_format</c> for the model to use, enabling "JSON mode" for guaranteed valid output.
-/// </summary>
-/// <remarks>
-/// <b>Important</b>: when using JSON mode, the model <b><u>must</u></b> also be instructed to produce JSON via a
-/// <c>system</c> or <c>user</c> message.
-/// <para>
-/// Without this paired, message-based accompaniment, the model may generate an unending stream of whitespace until the
-/// generation reaches the token limit, resulting in a long-running and seemingly "stuck" request.
-/// </para>
-/// <para>
-/// Also note that the message content may be partially cut off if <c>finish_reason</c> is <c>length</c>, which
-/// indicates that the generation exceeded <c>max_tokens</c> or the conversation exceeded the max context length for
-/// the model.
-/// </para>
-/// </remarks>
-[CodeGenModel("CreateChatCompletionRequestResponseFormat")]
-public partial class ChatResponseFormat
+[CodeGenModel("ChatResponseFormat")]
+public abstract partial class ChatResponseFormat
```

Comment:
I think this needs to implement `IEquatable<ChatResponseFormat>`

Reply:
Added to both; also did `IEquatable<string>` for the Assistants version to make `AssistantResponseFormat.CreateAutoFormat(), Is.EqualTo("auto")` work, too. Smoke test coverage added.